src/connection_manager.c

```diff
@@ -176,6 +176,46 @@ static void set_config_host(qd_server_config_t *config, qd_entity_t* entity)
     assert(config->host);
 }
 
+static char* qd_config_ssl_profile_get_password(qd_config_ssl_profile_t* ssl_profile)
+{
+    char *pw = ssl_profile->ssl_password;
+    if (!pw) return pw;
+
+    /* if the "password" starts with "env:" or "env: " then the remaining
+     * text is the environment variable that contains the password
+     */
+    if (strncmp(pw, "env:", 4) == 0) {
+        char *env = pw + 4;
+        /* skip the space if it is there */
+        if (*env == ' ') ++env;
+
+        const char* passwd = getenv(env);
+        if (passwd) {
+            free(ssl_profile->ssl_password);
+            pw = ssl_profile->ssl_password = strdup(passwd);
+        } else {
+            qd_error(QD_ERROR_NOT_FOUND, "Failed to find a password in the environment variable '%s'", env);
```

Comment:
Oh, you mean for the case where that isn't an env variable, but actually is part of a password?  Hmm, that's true.  though, it seems like it would help you understand what is happening faster if we log it.  Is the log file less secure than the config file?  Cuz, in that case, they already have the password in the config file.